main.go

```diff
@@ -110,6 +123,34 @@ func main() {
 		log.Printf("Overlapping routes %v with the range %s", routes, natV4Range)
 	}
 
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Cannot fetch cluster config: %v", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+
+	if len(podCIDR) == 0 {
+		log.Printf("Watching for node, awaiting podCIDR allocation")
+		ctx := context.Background()
+		node, err := waitForPodCIDR(ctx, k8sClient, hostname)
+		if err != nil {
+			log.Fatalf("waitForPodCIDR: %v", err)
+		}
+		// take podCIDR from primary IP family configured for the cluster
+		podCIDR = node.Spec.PodCIDRs[0]
+		log.Printf("podCIDR: %s", podCIDR)
+	}
+
+	_, _, podMaskSize, err := parsePodCIDR(podCIDR)
+	if err != nil {
+		log.Fatalf("Wrong pod CIDR %s: %v", podCIDR, err)
+	}
+	// TODO: it's fixed to /120 now because IPv4 range for source addresses for NAT is hardcoded
+	//       right now, this will be later changed to depend on the size of that range once it's configurable
+	if podMaskSize > 120 {
```

Comment:
mask arithmetic is the other way,no?
 mask 112 means 2^(128-112) possible host addresses that is larger than 2^(128-120)
```suggestion
	if podMaskSize < 120 {
```

Reply:
right :D covered that, needed to change kind config both in e2e and kind-ipv6.yaml, but tests are passing, so all seems good